flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/table/JdbcDynamicTableSourceITCase.java

```diff
@@ -368,37 +577,48 @@ private List<Row> executeQuery(String query) {
         // jdbc does support project push down, the cached row has been projected
-        RowData key1 = GenericRowData.of(1L);
-        RowData value1 =
-                GenericRowData.of(
-                        1L,
-                        DecimalData.fromBigDecimal(BigDecimal.valueOf(100.1234), 10, 4),
-                        TimestampData.fromLocalDateTime(
-                                truncateTime(LocalDateTime.parse("2020-01-01T15:35:00.123456"))));
-
-        RowData key2 = GenericRowData.of(2L);
-        RowData value2 =
-                GenericRowData.of(
-                        2L,
-                        DecimalData.fromBigDecimal(BigDecimal.valueOf(101.1234), 10, 4),
-                        TimestampData.fromLocalDateTime(
-                                truncateTime(LocalDateTime.parse("2020-01-01T15:36:01.123456"))));
-
-        RowData key3 = GenericRowData.of(3L);
-
-        Map<RowData, Collection<RowData>> expectedEntries = new HashMap<>();
-        expectedEntries.put(key1, Collections.singletonList(value1));
-        expectedEntries.put(key2, Collections.singletonList(value2));
-        expectedEntries.put(key3, Collections.emptyList());
+        LookupCacheAssert.assertThat(cache).containsExactlyEntriesOf(expectedCachedEntries);
+    }
 
-        LookupCacheAssert.assertThat(cache).containsExactlyEntriesOf(expectedEntries);
+    @Test
+    public void testResolvingParams() {
+
+        Map<String, String> properties = JdbcDynamicTableFactoryTest.getAllOptions();
+        properties.put("driver", "org.apache.derby.jdbc.EmbeddedDriver");
+        properties.put("username", "user");
+        properties.put("password", "pass");
+        properties.put("connection.max-retry-timeout", "120s");
+
+        // validation for source
+        DynamicTableSource testSource = createTableSource(SCHEMA, properties);
+        InternalJdbcConnectionOptions options =
+                InternalJdbcConnectionOptions.builder()
+                        .setDBUrl("jdbc:derby:memory:mydb")
+                        .setTableName("mytable")
+                        .setDriverName("org.apache.derby.jdbc.EmbeddedDriver")
+                        .setUsername("user")
+                        .setPassword("pass")
+                        .setConnectionCheckTimeoutSeconds(120)
+                        .build();
+        String actual =
+                ((JdbcDynamicTableSource) testSource).resolvePredicateParam("'type'=?", "111");
+
+        assert (actual.equals("'type'=111"));
+        // check column name containing a ?
+        actual = ((JdbcDynamicTableSource) testSource).resolvePredicateParam("'type?'=?", "111");
+        assert (actual.equals("'type?'=111"));
+        // param with lIKE string
+        actual =
+                ((JdbcDynamicTableSource) testSource)
+                        .resolvePredicateParam("'type?'=?", "LIKE 'a?'");
+        assert (actual.equals("'type?'=LIKE 'a?'"));
```

Comment:
it's better to use assertj to follow same way as other code in tests

Reply:
done